lib/internal/Magento/Framework/View/Asset/Repository.php

```diff
@@ -146,7 +146,7 @@ public function updateDesignParams(array &$params)
         }
 
         if ($theme) {
-            $params['themeModel'] = $this->getThemeProvider()->getThemeByFullPath($area . '/' . $theme);
+            $params['themeModel'] = $this->getThemeProvider()->getThemeById($theme);
```

Comment:
I changed base branch, please force push into the same branch applying your commit on top of `develop`.

Yeah, you are right about line 145.

Just finish your implementation without integration test then.